tryingsnake/__init__.py

```diff
@@ -247,6 +247,9 @@ def __eq__(self, other: Any) -> bool:
         return (isinstance(other, Success) and
                 self._v == other._v)
 
+    def __hash__(self):
```

Comment:
What about cases when  `_v` is not hashable? Are we fine with `TypeError` here?  Should result be `None`?

In Python hashing is, by convention, associated with immutable objects. In Scala from the other hand, mutable objects provide hashes, which are either meaningless (i.e. `Array`) or change over lifetime (like `scala.collection.mutable`).

Reply:
Good question! :) 
 
if `_v` is indeed mutable I think the TypeError raised by hash is good enough to detect the error. My reasoning is as follows: if I can use `x` as a key for a dict (i.e. is hashable and has the `__eq__` dunde defined), then I should be able to use `Success(x)/Failure(e)` as well. By default, all exceptions in Python are hashable.

In my case, I want to use a immutable object as key of a mapping, however that object has a `Try_`  instance as part of the member variable.

What do you think?.

Reply:
The use case looks perfectly reasonable - no doubt here.

As of depending on `TypeError` thrown by `hash` - that's what immutable containers (like `tuple`) do when mutable value is encountered. So this seems sensible and idiomatic as well.

The only thing I'd consider is re-throwing the exception, i.e.

```python
def __hash__(self):
    try:
        return hash(self._v)
    except TypeError as e:
        raise TypeError("Cannot hash try with unhashable value") from e
```

to be explicit, but I guess that it might be an overkill.

Reply:
That sounds sensible to me, I will make the changes.